pkg/store/bucket.go

```diff
@@ -3576,6 +3570,8 @@ func (r *bucketChunkReader) loadChunks(ctx context.Context, res []seriesEntry, a
 // save saves a copy of b's payload to a memory pool of its own and returns a new byte slice referencing said copy.
 // Returned slice becomes invalid once r.block.chunkPool.Put() is called.
 func (r *bucketChunkReader) save(b []byte) ([]byte, error) {
+	r.mtx.Lock()
```

Comment:
Then we can rename `r.mtx` to `r.chunkBytesMtx`.

Reply:
If we don't embed a lock to `queryStats` I will keep the mutex here